daemon/openrazer_daemon/hardware/device_base.py

```diff
@@ -104,13 +134,371 @@ def __init__(self, device_path, device_number, config, testing=False, additional
             ('razer.device.misc', 'hasDedicatedMacroKeys', self.dedicated_macro_keys, None, 'b'),
         }
 
+        effect_methods = {
+            ('razer.device.misc', 'getCurrentEffect', self.get_current_effect, None, 's'),
+            ('razer.device.misc', 'getCurrentEffectColors', self.get_current_effect_colors, None, 'ay'),
+            ('razer.device.misc', 'getCurrentEffectSpeed', self.get_current_effect_speed, None, 'i'),
+            ('razer.device.misc', 'getCurrentWaveDir', self.get_current_wave_dir, None, 'i'),
+        }
+
+        logo_effect_methods = {
+            ('razer.device.misc', 'getCurrentLogoEffect', self.get_current_logo_effect, None, 's'),
+            ('razer.device.misc', 'getCurrentLogoEffectColors', self.get_current_logo_effect_colors, None, 'ay'),
+            ('razer.device.misc', 'getCurrentLogoEffectSpeed', self.get_current_logo_effect_speed, None, 'i'),
+        }
+
+        scroll_effect_methods = {
+            ('razer.device.misc', 'getCurrentScrollEffect', self.get_current_scroll_effect, None, 's'),
+            ('razer.device.misc', 'getCurrentScrollEffectColors', self.get_current_scroll_effect_colors, None, 'ay'),
+            ('razer.device.misc', 'getCurrentScrollEffectSpeed', self.get_current_scroll_effect_speed, None, 'i'),
+        }
+
+        left_effect_methods = {
+            ('razer.device.misc', 'getCurrentLeftEffect', self.get_current_left_effect, None, 's'),
+            ('razer.device.misc', 'getCurrentLeftEffectColors', self.get_current_left_effect_colors, None, 'ay'),
+            ('razer.device.misc', 'getCurrentLeftEffectSpeed', self.get_current_left_effect_speed, None, 'i'),
+        }
+
+        right_effect_methods = {
+            ('razer.device.misc', 'getCurrentRightEffect', self.get_current_right_effect, None, 's'),
+            ('razer.device.misc', 'getCurrentRightEffectColors', self.get_current_right_effect_colors, None, 'ay'),
+            ('razer.device.misc', 'getCurrentRightEffectSpeed', self.get_current_right_effect_speed, None, 'i'),
+        }
+
         for m in methods:
             self.logger.debug("Adding {}.{} method to DBus".format(m[0], m[1]))
             self.add_dbus_method(m[0], m[1], m[2], in_signature=m[3], out_signature=m[4])
 
+        if 'set_static_effect' in self.METHODS:
```

Comment:
This could break on some older devices that just have on/off and no rgb (see Abyssus 1800 for example)

Reply:
This effect storage pull request only supports Chroma devices and doesn't work for devices with the set_logo_active method.
(However, I can add support for it although it'll require testing since I don't own any devices from that era)

Reply:
Done with last set of commits that implement storage for LED status.